dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/header/ReconnectTimerTask.java

```diff
@@ -42,15 +42,17 @@ protected void doTask(Channel channel) {
             Long lastRead = lastRead(channel);
             Long now = now();
             if (lastRead != null && now - lastRead > heartbeatTimeout) {
-                logger.warn("Close channel " + channel + ", because heartbeat read idle time out: "
-                        + heartbeatTimeout + "ms");
                 if (channel instanceof Client) {
                     try {
+                        logger.warn("Reconnect to remote channel " + channel.getRemoteAddress() + ", because heartbeat read idle time out: "
+                                + heartbeatTimeout + "ms");
                         ((Client) channel).reconnect();
                     } catch (Exception e) {
                         //do nothing
```

Comment:
should we log a warn message here too, as reconnect has encounter a exception here?

Reply:
yes,I have completely separated the reconnect log from the close log